src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -3858,22 +3869,46 @@ export class ResultsViewPageStore {
         },
     });
 
-    readonly treatmentsInStudies = remoteData<Treatment[]>({
+    readonly genericAssayEntitiesGroupByGenericAssayType = remoteData<{
+        [genericAssayType: string]: GenericAssayMeta[];
+    }>({
         await: () => [this.molecularProfilesInStudies],
         invoke: async () => {
-            return await fetchTreatmentByMolecularProfileIds(
+            return await fetchGenericAssayMetaByMolecularProfileIdsGroupByGenericAssayType(
                 this.molecularProfilesInStudies.result
             );
         },
     });
 
-    readonly selectedTreatments = remoteData<Treatment[]>({
-        await: () => [this.treatmentsInStudies],
+    readonly genericAssayEntitiesGroupByMolecularProfileId = remoteData<{
+        [genericAssayType: string]: GenericAssayMeta[];
+    }>({
+        await: () => [this.molecularProfilesInStudies],
+        invoke: async () => {
+            return await fetchGenericAssayMetaByMolecularProfileIdsGroupByMolecularProfileId(
+                this.molecularProfilesInStudies.result
+            );
+        },
+    });
+
+    readonly selectedGenericAssayEntitiesGroupByGenericAssayType = remoteData<{
+        [genericAssayType: string]: GenericAssayMeta[];
+    }>({
+        await: () => [this.genericAssayEntitiesGroupByGenericAssayType],
         invoke: () => {
-            const treatmentIdFromUrl = this.treatmentList;
+            const selectedGenericAssayEntitiesFromUrl: {
```

Comment:
Strange that you have to type this variable.   Should be inferred from parsedGenericAssayGroups.

Also, why do we even bother assigning to local variable rather than just referencing directly.   I actuallyt think "selectedGenericAssayEntities" is a good name for this computed property.